ogre2/src/Ogre2GpuRays.cc

```diff
@@ -201,20 +201,63 @@ class IGNITION_RENDERING_OGRE2_HIDDEN ignition::rendering::Ogre2GpuRaysPrivate
 using namespace ignition;
 using namespace rendering;
 
-// Arbitrary value
+// Arbitrary values
 static const uint32_t kLaserRetroMainDepthPassId = 9525u;
+static const uint32_t kLaserRetro1stPassQuad = 9526u;
+
+/// \brief standard deviation of particle noise
+static const double kParticleStddev = 0.01;
 
 //////////////////////////////////////////////////
 Ogre2LaserRetroMaterialSwitcher::Ogre2LaserRetroMaterialSwitcher(
-    Ogre2ScenePtr _scene)
+  Ogre2ScenePtr _scene, Ogre2GpuRays *_gpuRays, Ogre::Camera *_ogreCamera)
 {
   this->scene = _scene;
+  this->gpuRays = _gpuRays;
+  this->ogreCamera = _ogreCamera;
 }
 
 //////////////////////////////////////////////////
 void Ogre2LaserRetroMaterialSwitcher::passPreExecute(
   Ogre::CompositorPass *_pass)
 {
+  if(_pass->getDefinition()->mIdentifier == kLaserRetro1stPassQuad)
+  {
+    IGN_ASSERT(dynamic_cast<Ogre::CompositorPassQuad *>(_pass),
+               "Impossible! Corrupted memory?");
+    Ogre::CompositorPassQuad *passQuad =
+      static_cast<Ogre::CompositorPassQuad *>(_pass);
+
+    Ogre::Pass *pass = passQuad->getPass();
+    Ogre::GpuProgramParametersSharedPtr psParams =
+      pass->getFragmentProgramParameters();
+
+    // Set the uniform variables (see gpu_rays_1st_pass_fs.glsl).
+    // The projectParams is used to linearize depth buffer data
+    // The other params are used to clamp the range output
+    Ogre::Vector2 projectionAB = this->ogreCamera->getProjectionParamsAB();
+    double projectionA = projectionAB.x;
+    double projectionB = projectionAB.y;
+    projectionB /= this->gpuRays->FarClipPlane();
+    psParams->setNamedConstant("projectionParams",
+                               Ogre::Vector2(static_cast<float>(projectionA),
+                                             static_cast<float>(projectionB)));
+    psParams->setNamedConstant(
+      "near", static_cast<float>(this->gpuRays->NearClipPlane()));
+    psParams->setNamedConstant(
+      "far", static_cast<float>(this->gpuRays->FarClipPlane()));
+    psParams->setNamedConstant("max",
+                               static_cast<float>(this->gpuRays->dataMaxVal));
+    psParams->setNamedConstant("min",
+                               static_cast<float>(this->gpuRays->dataMinVal));
+    psParams->setNamedConstant("particleStddev",
+                               static_cast<float>(kParticleStddev));
```

Comment:
Cannot do. Doing so breaks 80-line column.

But I edited to have local variables instead, so that it fits (though now the whole line fits in a single line...).

This would be much easier if Gazebo just enforced mandatory `clang-format` (that's for another discussion).

Reply:
Done.